solidity/test/token_stake/TestPunishment.js

```diff
@@ -0,0 +1,235 @@
+            let amountToSlash = web3.utils.toBN(1000)
+            await expectRevert(
+                stakingContract.slash(amountToSlash, [operator], { from: operatorContract }),
+                "Operator stake must be active"
+            )
+        })
+
+        it("should fail when operator stake is released", async () => {
+            time.increase((await stakingContract.initializationPeriod()).addn(1))
+            await stakingContract.undelegate(operator, { from: owner })
+            time.increase((await stakingContract.undelegationPeriod()).addn(1))
+
+            let amountToSlash = web3.utils.toBN(100);
+            await expectRevert(
+                stakingContract.slash(amountToSlash, [operator], { from: operatorContract }),
+                "Stake is released"
+            )
+        })
+    })
+
+    describe("seize", () => {
+        it("should seize token amount from stake", async () => {
+            time.increase((await stakingContract.initializationPeriod()).addn(1))
+
+            let operatorBalanceBeforeSeizing = await stakingContract.balanceOf(operator)
+            let tattletaleBalanceBeforeSeizing = await token.balanceOf(tattletale)
+
+            let amountToSeize = web3.utils.toBN(42000000)
+            let rewardMultiplier = web3.utils.toBN(25)
+            await stakingContract.seize(
+                amountToSeize, rewardMultiplier, tattletale,
+                [operator], { from: operatorContract }
+            )
+
+            let operatorBalanceAfterSeizing = await stakingContract.balanceOf(operator)
+            let tattletaleBalanceAfterSeizing = await token.balanceOf(tattletale)
+
+            expect(operatorBalanceAfterSeizing).to.eq.BN(
+                operatorBalanceBeforeSeizing.sub(amountToSeize)
+            )
+
+            // 525000 = (42000000 * 5 / 100) * 25 / 100
+            let expectedTattletaleReward = web3.utils.toBN(525000)
+            expect(tattletaleBalanceAfterSeizing).to.eq.BN(
+                tattletaleBalanceBeforeSeizing.add(expectedTattletaleReward)
+            )
+        })
+
+        it("should seize no more than available on stake", async () => {
+            time.increase((await stakingContract.initializationPeriod()).addn(1))
+
+            let tattletaleBalanceBeforeSeizing = await token.balanceOf(tattletale)
+
+            let amountToSeize = largeStake.add(web3.utils.toBN(100)) // 200000000000000000000100
+            let rewardMultiplier = web3.utils.toBN(10)
+            await stakingContract.seize(
+                amountToSeize, rewardMultiplier, tattletale,
+                [operator], { from: operatorContract }
+            )
+
+            let operatorBalanceAfterSeizing = await stakingContract.balanceOf(operator)
+            let tattletaleBalanceAfterSeizing = await token.balanceOf(tattletale)
+
+            expect(operatorBalanceAfterSeizing).to.eq.BN(0)
+
+            // 1000000000000000000000 = (200000000000000000000100 * 5 / 100) * 10 / 100
```

Comment:
Should we test with a higher excess to ensure that the tattletale reward is calculated from the applied penalty, not the requested penalty?

Reply:
Yep, very good point. Fixed in bbbe3dd2255.